pam/internal/adapter/gdmmodel.go

```diff
@@ -0,0 +1,283 @@
+package adapter
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/msteinert/pam/v2"
+	"github.com/ubuntu/authd"
+	"github.com/ubuntu/authd/internal/brokers/responses"
+	"github.com/ubuntu/authd/internal/log"
+	"github.com/ubuntu/authd/pam/internal/gdm"
+	"github.com/ubuntu/authd/pam/internal/proto"
+)
+
+const (
+	gdmPollFrequency time.Duration = time.Millisecond * 16
+)
+
+type gdmModel struct {
+	pamMTx pam.ModuleTransaction
+
+	waitingAuth bool
+}
+
+type gdmPollDone struct{}
+
+func newGdmModel(mTx pam.ModuleTransaction) gdmModel {
```

Comment:
It‘s called only once and there is no side-effect by this function, no? So I think build directly the struct on the caller side.

Reply:
Ok, it was just for consistency with other models that are all initialized with a function, so I followed the pattern, but I can fix this.